service/migration_manager.hh

```diff
@@ -193,6 +195,17 @@ public:
     void set_concurrent_ddl_retries(size_t);
 };
 
+extern template
+future<> migration_manager::announce_with_raft<schema_change>(std::vector<mutation> schema, group0_guard, std::string_view description);
+extern template
+future<> migration_manager::announce_with_raft<topology_change>(std::vector<mutation> schema, group0_guard, std::string_view description);
+
+extern template
+future<> migration_manager::announce<schema_change>(std::vector<mutation> schema, group0_guard, std::string_view description);
+extern template
+future<> migration_manager::announce<topology_change>(std::vector<mutation> schema, group0_guard, std::string_view description);
```

Comment:
let's find a better way to handle the different types than this template. It's a beginning of a nightmare that we should avoid.

Reply:
These explicit instantiation calls are here to only produce 1 instance of a template for a given type, otherwise every translation unit would instantiate and compile its own version of this function.
The simplest would be to remove all of this and just leave a template in a header file, but having compilation time in mind, I wanted to not add an extra overhead.

Reply:
Referenced in https://github.com/scylladb/scylladb/issues/1#issuecomment-2115788825

Reply:
I'll close the comment, but please feel free to reopen.

Reply:
Added to https://github.com/scylladb/scylladb/issues/19126